db/crud.go

```diff
@@ -400,6 +400,14 @@ func (db *Database) GetDelta(docID, fromRevID, toRevID string) (delta []byte, er
 		}
 		// We didn't copy fromBody earlier (in case we could get by with just the delta), so need do it now
 		fromBodyCopy := fromRevision.Body.DeepCopy()
+
+		// If attachments have changed between these revisions, we'll stamp the metadata into the bodies before diffing
+		// so that the resulting delta also contains attachment metadata changes
+		if !fromRevision.Attachments.Equal(toBody.Attachments) {
```

Comment:
The work being done in Equal doesn't feel particularly more efficient than the work Diff is going to do with the _attachments struct, and has the downside of being performed twice when there is a difference found.  

It feels like it would be more efficient to just add the attachments to fromBodyCopy/toBody.Body when their respective Attachments objects are non-empty.